enhancements/agent-installer/simplified-registry-less-cluster-operations.md

```diff
@@ -0,0 +1,426 @@
+   configuration).
+2. The user attaches the newly generate node ISO to the target machine(s),
+   and boots it. The progress and status of the task can be monitored via the
+   `oc adm node-image monitor` command.
+3. The user waits until the CSRs are generated, and manually approves them
+   via the `oc adm certificate approve` command.
+   
+### API Extensions
+
+#### InternalReleaseImage custom resource
+
+The `InternalReleaseImage` is a new singleton custom resource, controlled by
+MCO, used to keep track and manage the OCP release images stored into the
+control planes nodes. In particular, it will be used for:
+
+* Triggering a new release image copy task - as a preliminary upgrade step
+* Delete from disk a release image version not anymore in use - as an optional
+  post-upgrade step
+* Enabled/disable the registry services
+
+This is an example of how the CR will look like:
+
+```
+apiVersion: releases.openshift.io/v1alpha1
+kind: InternalReleaseImage
+metadata:
+  name: cluster
+spec:
+  releases:
+    - "quay.io/openshift-release-dev/ocp-release:4.18.0-x86_64"
+    - "quay.io/openshift-release-dev/ocp-release:4.19.0-x86_64"
+  enabled: true
+status:
+  releases:
+    - "quay.io/openshift-release-dev/ocp-release:4.18.0-x86_64"
+    - "quay.io/openshift-release-dev/ocp-release:4.19.0-x86_64"
+  enabled: true
+  conditions:
+    - type: "Available"
+      status: "True"
+      lastTransitionTime: "2025-07-12T14:02:00Z"
+      reason: "ReleasesSynchronized"
+      message: "The previously managed releases are available."
+    - type: "Progressing"
+      status: "True"
+      lastTransitionTime: "2025-07-12T13:13:00Z"
+      reason: "AddingRelease"
+      message: "Adding new release image quay.io/openshift-release-dev/ocp-release:4.19.0-x86_64 on master-0."
+  ...
+```
+
+* The `spec.releases` will be used to configure the desired release contents.
+  Adding (or removing) an entry from the list will trigger an update on all the
+  control plane nodes.
+* The `spec.enabled` will be used to turn on / off the feature. In particular,
+  when turned off, all the registry services running on the control plane nodes
+  will be stopped.
+* The `status.releases` field will report the currently managed releases
+* The `status.conditions` field will be used to keep track of the activity
+  performed on the various nodes.
+
+### Topology Considerations
+
+Will function identically in all topologies.
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+#### Extended RHCOS ISO build
+
+The release image and operators images will be stored inside the extended RHCOS
+ISO in the storage format used by the [distribution/distribution](https://github.com/distribution/distribution)
+registry’s filesystem storage plugin.
+
+#### Distribution/distribution registry OCP integration
```

Comment:
The image registry here has some interesting overlap with the MCO's image mode efforts, namely:
1. the design seems to be a lightweight, cluster-only registry used for cluster operations that we manage (somewhat through the MCO)
2. this registry could potentially be used for os image build distribution for this cluster

cc @pablintino who has been investigating image registry use cases for image mode